internal/dataplane/fallback/fallback.go

```diff
@@ -31,25 +31,25 @@ func NewGenerator(cacheGraphProvider CacheGraphProvider, logger logr.Logger) *Ge
 func (g *Generator) GenerateExcludingAffected(
 	cache store.CacheStores,
 	brokenObjects []ObjectHash,
-) (store.CacheStores, error) {
+) (store.CacheStores, store.SnapshotHash, error) {
```

Comment:
Nit: godoc and arg mentions `broken` but the name of this functions uses `Affected` instead. I'd suggest that we're consistent and pick one of those.

---

Godoc should be updated to include the description of the added return value.